frontend/catalyst/passes.py

```diff
@@ -371,34 +365,37 @@ def circuit(x: float):
     >>> circuit(0.54)
     Array(0.5965506257017892, dtype=float64)
     """
-    if not isinstance(fn, qml.QNode):
-        raise TypeError(f"A QNode is expected, got the classical function {fn}")
+    if not isinstance(qnode, qml.QNode):
+        raise TypeError(f"A QNode is expected, got the classical function {qnode}")
 
-    funcname = fn.__name__
-    wrapped_qnode_function = fn.func
+    clone = copy.copy(qnode)
+    clone.__name__ += "_merge_rotations"
 
-    def wrapper(*args, **kwrags):
-        if EvaluationContext.is_tracing():
-            apply_registered_pass_p.bind(
-                pass_name="merge-rotations",
-            )
-        return wrapped_qnode_function(*args, **kwrags)
+    @functools.wraps(clone)
+    def wrapper(*args, **kwargs):
+        pass_pipeline = kwargs.pop("pass_pipeline", tuple())
+        pass_pipeline += (Pass("merge-rotations"),)
+        kwargs["pass_pipeline"] = pass_pipeline
+        return clone(*args, **kwargs)
 
-    fn_clone = copy.copy(fn)
-    fn_clone.func = wrapper
-    fn_clone.__name__ = funcname + "_merge_rotations"
-
-    return fn_clone
+    return wrapper
 
 
 def _API_name_to_pass_name():
     return {"cancel_inverses": "remove-chained-self-inverse", "merge_rotations": "merge-rotations"}
 
 
-def _add_mlir_quantum_decomposition(device):
-    """When called it adds the MLIR decomposition pass thanks to the transform dialect."""
-    # TODO: make this non related to the name of the device
-    if device.original_device.name == "oqd.cloud":
-        apply_registered_pass_p.bind(
-            pass_name="ions-decomposition",
-        )
+def ions_decomposition(qnode=None):
+    """Apply decomposition pass at the MLIR level"""
+
+    if not isinstance(qnode, qml.QNode):
+        raise TypeError(f"A QNode is expected, got the classical function {qnode}")
+
+    @functools.wraps(qnode)
+    def wrapper(*args, **kwargs):
+        pass_pipeline = kwargs.pop("pass_pipeline", tuple())
+        pass_pipeline += (Pass("ions-decomposition"),)
+        kwargs["pass_pipeline"] = pass_pipeline
+        return qnode(*args, **kwargs)
+
+    return wrapper
```

Comment:
Mmm... I know for a fact that it is tested on the lit tests already. I am generally ok with increasing coverage but I didn't want to just add it. Since you are suggesting it, I will add it :)

Reply:
Mmm... changed my mind, the lit test does a little bit more than just running the pass. If you insist, I will add it.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1349/commits/eb25823e764e114005cb4b644a348f69ba5e4068

Reply:
Works with me, as long as we have the lit test I think it's fine.